### ISSUE TITLE
`:` autocomplete doesn't work if code follows

### ISSUE THREAD

**Issue by PennyJim** [EmmyLuaLs/emmylua-analyzer-rust#1144]
Trying to eek out `:` autocomplete suggestions works fine when at the end of the file, but the moment you're inserting code into previously written code it refuses to give any suggestions. Whereas `.` suggestions still work fine.
```lua
---@class B
local B = {}
function B:one() return self end

do
	B: -- No suggesstions.
end

do
	B.  -- `one(self)` suggested
end
```

Screenshots for proof:

<img width="374" height="181" alt="Image" src="https://github.com/user-attachments/assets/50510e0b-5e42-42b0-aa1f-3cd9ae184214" />
<img width="374" height="181" alt="Image" src="https://github.com/user-attachments/assets/b95a235d-3080-49cd-84f1-8ceabfb07d07" />

Just in case autocomplete could be dropping out at any point in this chain: 
VSCodium version: 1.121.03429
EmmyLua Extension: xuhuanzy.emmylua-luals v0.9.38
EmmyLua version: xuhuanzy's update branch

**Comment by PennyJim** [comment]
Man.. I simplified the case to the point where the fix didn't catch half the bug.
The reason I used a self-returning function was because that's what I'm trying to use it on.

With @xuhuanzy's https://github.com/xuhuanzy/emmylua-ng/commit/923e80b1a867496dbaf8e58779751045ce4e8e83, I'm still having the issue in this case:
```lua
do
	B:one():
end
```

**Comment by xuhuanzy** [comment]
fixed 05990bcd423096ed911648b41f3a751f37709ae3

**Comment by PennyJim** [comment]
From the tests, it looks like the scope boundaries are entirely handled
But it still does not generically handle any expression following it
```lua
B:one():<??>
var = B:one():<??>
local var = B:one():<??>
B.one(B)
```

My use case is mostly within table definitions, and it works if followed by a `,` so it's not urgent (as if it was before). Just not entirely fixed

What interests me is that if you replace `:` with a `.` in each instance, it works fine. So I feel like whatever the difference in handling between the index and the colon is the root of the problem.

**Comment by PennyJim** [comment]
I may not know much about the structure of the project, but it might help to point out that an incomplete index with `.` complains about expecting a field name, while `:` says the colon is unexpected.

I wonder if changing the ast/parser might help fix the problem here.

**Comment by xuhuanzy** [comment]
This is an issue introduced when supporting luajit-extension/luajit3.x. A simple solution is to disable filtering when ternary expressions are not supported.

@CppCXY 具体你自己修得了
